lib/rpmsg/rpmsg.c

```diff
@@ -259,17 +259,12 @@ int rpmsg_init_vdev(struct rpmsg_virtio_device *rvdev,
 
 		if (!rvdev->shbuf)
 			return RPMSG_ERR_NO_MEM;
-
-		vq_names[0] = "tx_vq";
-		vq_names[1] = "rx_vq";
-		callback[0] = rpmsg_tx_callback;
-		callback[1] = rpmsg_rx_callback;
-	} else {
-		vq_names[0] = "rx_vq";
-		vq_names[1] = "tx_vq";
-		callback[0] = rpmsg_rx_callback;
-		callback[1] = rpmsg_tx_callback;
 	}
+
+	vq_names[0] = "rx_vq";
```

Comment:
i think this potentially introduce a bug. vrings are crossed between the both processors.
master-> vring0 ->slave
master <- vring1<- slave

rpmsg_virtio_create_virtqueues calle  after  parse the vdev->vrings_info associated...

Reply:
I understand, will try and fix this up.

I think we should re-do how we associate the vring, virtqueue's and init it all.